clients/admin-ui/src/features/datastore-connections/add-connection/forms/YamlEditorForm.tsx

```diff
@@ -93,7 +93,12 @@ const YamlEditorForm: React.FC<YamlEditorFormProps> = ({
     // Only need the confirmation if we are overwriting, which only happens when
     // there is already data
     if (data.length) {
-      warningDisclosure.onOpen();
+      const value: string = (monacoRef.current as any).getValue();
+      // Check if the fides key that is in the editor is the same as the one that already exists
+      // If so, then it is an overwrite and we should open the confirmation modal
+      if (data.some((d) => value.includes(`fides_key: ${d.fides_key}\n`))) {
+        warningDisclosure.onOpen();
+      }
```

Comment:
I think the check is really does this `fides_key` match any fides_key in the ctl_datasets table? Because we could still update it to match a different ctl_dataset and that would be even more unintuitive. 

Regardless, if you update the key, it now looks like you don't have an option to save anything at all, you just click the save button and nothing happens.   It's probably safer if they can't update the fides_key without a warning, but the UI is in a weird state still.

Reply:
ah I see, okay yes there are two things going on here, thanks for catching that!